src/main/java/twentymethods/TwentyMethods.java

```diff
@@ -123,4 +125,83 @@ public static int findIndexOfMatchingIntValueInArrayStartingFromArrayEnd(int[] i
         }
         return -1;
     }
+
+    //10.1) метод принимает инт, и возвращает факториал от заданого инта \
+    public static int getFactorialFromIntFirst(int number) {
+        int result = 1;
+        for (int i = 1; i <= number; i++) {
+            result = result * i;
+        }
+        return result;
+    }
+
+    //10.2) метод принимает инт, и возвращает факториал от заданого инта \
+    public static int getFactorialFromIntSecond(int n) {
+        int result;
+        if (n == 1)
+            return 1;
+        result = getFactorialFromIntSecond(n - 1) * n;
+        return result;
+    }
+
+    //11) принимает инт год, и возвращает тру если год высокосный
+    public static boolean leapCheckYears(int year) {
+        return ((year % 4 == 0) || (year % 100 == 0));
```

Comment:
not full formula, please edit
"Each leap year has 366 days instead of 365. This extra leap day occurs in each year that is a multiple of 4 (except for years evenly divisible by 100, but not by 400)"